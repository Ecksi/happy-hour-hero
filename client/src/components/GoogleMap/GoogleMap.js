import React, { Component } from 'react';
import { connect } from 'react-redux';
import GoogleMapReact from 'google-map-react';
import './GoogleMap.css';
import UserMarker from '../UserMarker/UserMarker';
import RestaurantMarker from '../RestaurantMarker/RestaurantMarker';
import PropTypes from 'prop-types';

export class GoogleMap extends Component {
  constructor() {
    super();

    this.state = {
      zoom: 16,
      todaysRestaurants: []
    };
  }

  componentDidUpdate(prevProps) {
    if (this.props.happyHours !== prevProps.happyHours) {
      this.todaysHappyHourRestaurants();
    }
  }

  componentDidMount() {
    if (this.props.happyHours) {
      this.todaysHappyHourRestaurants();
    }
  }

  todaysHappyHourRestaurants = () => {
    const todaysRestaurants = this.props.filteredRestaurants.reduce((restaurants, restaurant) => {
      this.props.happyHours.forEach((happyHour) => {
        if (happyHour.restaurant_id === restaurant.id && happyHour.day === this.props.day && !restaurants.includes(restaurant)) {
          restaurants.push(restaurant);
        }
      });

      return restaurants;
    }, []);

    this.setState({todaysRestaurants});
  }

  render() {
    let markers;

    const center = {
      lat: this.props.location.latitude,
      lng: this.props.location.longitude
    };

    if (this.state.todaysRestaurants) {
      markers = this.state.todaysRestaurants.map((restaurant, index) => {
        const { latitude, longitude, name, id } = restaurant;

        return (<RestaurantMarker
          lat={ latitude }
          lng={ longitude }
          key={ index }
          name={ name }
          id={ id }
        />);
      });

    }

    return (
      <div className='google-map'>
        <GoogleMapReact
          center={ center }
          defaultZoom={ this.state.zoom }>
          <UserMarker
            lat={ center.lat }
            lng={ center.lng }
          />
          { markers }
        </GoogleMapReact>
      </div>
    );
  }
}

GoogleMap.propTypes = {
  location: PropTypes.object,
  filteredRestaurants: PropTypes.array,
};

export const mapStateToProps = (state) => ({
  location: state.location,
  filteredRestaurants: state.filteredRestaurants,
  happyHours: state.happyHours,
  day: state.day
});

export default connect(mapStateToProps)(GoogleMap);
