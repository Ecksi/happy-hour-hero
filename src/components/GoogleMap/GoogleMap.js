import React, { Component } from 'react';
import { connect } from 'react-redux';
import GoogleMapReact from 'google-map-react';
import './GoogleMap.css';
import UserMarker from '../UserMarker/UserMarker';
import RestaurantMarker from '../RestaurantMarker/RestaurantMarker';

class GoogleMap extends Component {
  constructor() {
    super();

    this.state = {
      zoom: 11
    };
  }
  
  render() {
    const center = {
      lat: this.props.location.latitude,
      lng: this.props.location.longitude
    };

    return (
      <div className='google-map'>
        <GoogleMapReact
          defaultCenter={ center }
          defaultZoom={ this.state.zoom }>
          <UserMarker
            lat={ center.lat }
            lng={ center.lng }
          />
          <RestaurantMarker
            lat={ center.lat + 0.02 }
            lng={ center.lng + 0.02 }
          />
        </GoogleMapReact>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  location: state.location
});

export default connect(mapStateToProps)(GoogleMap);