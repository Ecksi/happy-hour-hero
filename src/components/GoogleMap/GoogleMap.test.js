import React from 'react';
import { shallow } from 'enzyme';
import GoogleMap from './GoogleMap';

describe('GoogleMap', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<GoogleMap />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});