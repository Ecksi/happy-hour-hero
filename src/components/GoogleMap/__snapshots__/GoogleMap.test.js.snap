// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap matches the snapshot 1`] = `undefined`;
